<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>ggcoverage User Guide
 • ggcoverage</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="ggcoverage User Guide
">
<meta property="og:description" content="ggcoverage">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ggcoverage</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.7.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/ggcoverage.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><link href="ggcoverage_files/font-awesome-5.1.0/css/all.css" rel="stylesheet">
<link href="ggcoverage_files/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip></h1>
<p>ggcoverage User Guide</p>
                        <h4 class="author">Yabing Song</h4>
            <address class="author_afil">
      School of Life Sciences, Tsinghua University<br><a class="author_email" href="mailto:#"></a><a href="mailto:songyb18@mails.tsinghua.edu.cn" class="email">songyb18@mails.tsinghua.edu.cn</a>
      </address>
                  
            <h4 class="date">3 九月 2022</h4>
      
      
      <div class="hidden name"><code>ggcoverage.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>The goal of <code>ggcoverage</code> is to simplify the process of visualizing genome coverage. It contains functions to load data from BAM, BigWig or BedGraph files, create genome coverage plot, add various annotation to the coverage plot, including base and amino acid annotation, GC annotation, gene annotation, transcript annotation, ideogram annotation and peak annotation.</p>
    </div>
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h1>
<p><code>ggcoverage</code> is an R package distributed as part of the <a href="https://cran.r-project.org/">CRAN</a>. To install the package, start R and enter:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install via CRAN</span>
<span class="fu">install.package</span><span class="op">(</span><span class="st">"ggcoverage"</span><span class="op">)</span>

<span class="co"># install via Github</span>
<span class="co"># install.package("remotes")   #In case you have not installed it.</span>
<span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"showteeth/ggcoverage"</span><span class="op">)</span></code></pre></div>
<p>In general, it is <strong>recommended</strong> to install from <a href="https://github.com/showteeth/ggcoverage">Github repository</a> (update more timely).</p>
<p>Once <code>ggcoverage</code> is installed, it can be loaded by the following command.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"rtracklayer"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"graphics"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"ggcoverage"</span><span class="op">)</span></code></pre></div>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p><img src="../man/figures/ggcoverage.png" align="right" width="200"></p>
<p>The goal of <code>ggcoverage</code> is simplify the process of visualizing genome coverage. It contains three main parts:</p>
<ul>
<li>
<strong>Load the data</strong>: <code>ggcoverage</code> can load BAM, BigWig (.bw), BedGraph files from various NGS data, including WGS, RNA-seq, ChIP-seq, ATAC-seq, et al.</li>
<li><strong>Create genome coverage plot</strong></li>
<li>
<strong>Add annotations</strong>: <code>ggcoverage</code> supports six different annotations:
<ul>
<li>
<strong>Base and amino acid annotation</strong>: Visualize genome coverage at single-nucleotide level with bases and amino acids.</li>
<li>
<strong>GC annotation</strong>: Visualize genome coverage with GC content</li>
<li>
<strong>gene annotation</strong>: Visualize genome coverage across whole gene</li>
<li>
<strong>transcription annotation</strong>: Visualize genome coverage across different transcripts</li>
<li>
<strong>ideogram annotation</strong>: Visualize the region showing on whole chromosome</li>
<li>
<strong>peak annotation</strong>: Visualize genome coverage and peak identified</li>
</ul>
</li>
</ul>
<p><code>ggcoverage</code> utilizes <code>ggplot2</code> plotting system, so its usage is <strong>ggplot2-style</strong>!</p>
<hr>
</div>
<div id="rna-seq-data" class="section level1">
<h1 class="hasAnchor">
<a href="#rna-seq-data" class="anchor"></a>RNA-seq data</h1>
<div id="load-the-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-the-data" class="anchor"></a>Load the data</h2>
<p>The RNA-seq data used here are from <a href="https://bioconductor.org/packages/release/data/experiment/html/RNAseqData.HNRNPC.bam.chr14.html">Transcription profiling by high throughput sequencing of HNRNPC knockdown and control HeLa cells</a>, we select four sample to use as example: ERR127307_chr14, ERR127306_chr14, ERR127303_chr14, ERR127302_chr14, and all bam files are converted to bigwig file with <a href="https://deeptools.readthedocs.io/en/develop/">deeptools</a>.</p>
<p>Load metadata:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load metadata</span>
<span class="va">meta.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"RNA-seq"</span>, <span class="st">"meta_info.csv"</span>, package <span class="op">=</span> <span class="st">"ggcoverage"</span><span class="op">)</span>
<span class="va">sample.meta</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="va">meta.file</span><span class="op">)</span>
<span class="va">sample.meta</span>
<span class="co">#&gt;        SampleName    Type Group</span>
<span class="co">#&gt; 1 ERR127302_chr14 KO_rep1    KO</span>
<span class="co">#&gt; 2 ERR127303_chr14 KO_rep2    KO</span>
<span class="co">#&gt; 3 ERR127306_chr14 WT_rep1    WT</span>
<span class="co">#&gt; 4 ERR127307_chr14 WT_rep2    WT</span></code></pre></div>
<p>Load track files:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># track folder</span>
<span class="va">track.folder</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"RNA-seq"</span>, package <span class="op">=</span> <span class="st">"ggcoverage"</span><span class="op">)</span>
<span class="co"># load bigwig file</span>
<span class="va">track.df</span> <span class="op">=</span> <span class="fu"><a href="../reference/LoadTrackFile.html">LoadTrackFile</a></span><span class="op">(</span>track.folder <span class="op">=</span> <span class="va">track.folder</span>, format <span class="op">=</span> <span class="st">"bw"</span>,
                         meta.info <span class="op">=</span> <span class="va">sample.meta</span><span class="op">)</span>
<span class="co"># check data</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">track.df</span><span class="op">)</span>
<span class="co">#&gt;   seqnames    start      end score    Type Group</span>
<span class="co">#&gt; 1    chr14 21572751 21630650     0 KO_rep1    KO</span>
<span class="co">#&gt; 2    chr14 21630651 21630700     1 KO_rep1    KO</span>
<span class="co">#&gt; 3    chr14 21630701 21630800     4 KO_rep1    KO</span>
<span class="co">#&gt; 4    chr14 21630801 21657350     0 KO_rep1    KO</span>
<span class="co">#&gt; 5    chr14 21657351 21657450     1 KO_rep1    KO</span>
<span class="co">#&gt; 6    chr14 21657451 21663550     0 KO_rep1    KO</span></code></pre></div>
<p>Prepare mark region:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># create mark region</span>
<span class="va">mark.region</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>start<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">21678900</span>,<span class="fl">21732001</span>,<span class="fl">21737590</span><span class="op">)</span>,
                       end<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">21679900</span>,<span class="fl">21732400</span>,<span class="fl">21737650</span><span class="op">)</span>,
                       label<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"M1"</span>, <span class="st">"M2"</span>, <span class="st">"M3"</span><span class="op">)</span><span class="op">)</span>
<span class="co"># check data</span>
<span class="va">mark.region</span>
<span class="co">#&gt;      start      end label</span>
<span class="co">#&gt; 1 21678900 21679900    M1</span>
<span class="co">#&gt; 2 21732001 21732400    M2</span>
<span class="co">#&gt; 3 21737590 21737650    M3</span></code></pre></div>
<p>Load GTF file:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gtf.file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"used_hg19.gtf"</span>, package <span class="op">=</span> <span class="st">"ggcoverage"</span><span class="op">)</span>
<span class="va">gtf.gr</span> <span class="op">=</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/GFFFile-class.html">import.gff</a></span><span class="op">(</span>con <span class="op">=</span> <span class="va">gtf.file</span>, format <span class="op">=</span> <span class="st">'gtf'</span><span class="op">)</span></code></pre></div>
<hr>
</div>
<div id="basic-coverage" class="section level2">
<h2 class="hasAnchor">
<a href="#basic-coverage" class="anchor"></a>Basic coverage</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">basic.coverage</span> <span class="op">=</span> <span class="fu"><a href="../reference/ggcoverage.html">ggcoverage</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">track.df</span>, color <span class="op">=</span> <span class="st">"auto"</span>, 
                            mark.region <span class="op">=</span> <span class="va">mark.region</span>, range.position <span class="op">=</span> <span class="st">"out"</span><span class="op">)</span>
<span class="va">basic.coverage</span></code></pre></div>
<p><img src="../man/figures/README-basic_coverage-1.png" width="576" style="display: block; margin: auto;"></p>
<p>You can also change Y axis style:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">basic.coverage</span> <span class="op">=</span> <span class="fu"><a href="../reference/ggcoverage.html">ggcoverage</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">track.df</span>, color <span class="op">=</span> <span class="st">"auto"</span>, 
                            mark.region <span class="op">=</span> <span class="va">mark.region</span>, range.position <span class="op">=</span> <span class="st">"in"</span><span class="op">)</span>
<span class="va">basic.coverage</span></code></pre></div>
<p><img src="../man/figures/README-basic_coverage_2-1.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="add-gene-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#add-gene-annotation" class="anchor"></a>Add gene annotation</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">basic.coverage</span> <span class="op">+</span> 
  <span class="fu"><a href="../reference/geom_gene.html">geom_gene</a></span><span class="op">(</span>gtf.gr<span class="op">=</span><span class="va">gtf.gr</span><span class="op">)</span></code></pre></div>
<p><img src="../man/figures/README-gene_coverage-1.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="add-transcript-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#add-transcript-annotation" class="anchor"></a>Add transcript annotation</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">basic.coverage</span> <span class="op">+</span> 
  <span class="fu"><a href="../reference/geom_transcript.html">geom_transcript</a></span><span class="op">(</span>gtf.gr<span class="op">=</span><span class="va">gtf.gr</span>,label.vjust <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></code></pre></div>
<p><img src="../man/figures/README-transcript_coverage-1.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="add-ideogram" class="section level2">
<h2 class="hasAnchor">
<a href="#add-ideogram" class="anchor"></a>Add ideogram</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">basic.coverage</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_gene.html">geom_gene</a></span><span class="op">(</span>gtf.gr<span class="op">=</span><span class="va">gtf.gr</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_ideogram.html">geom_ideogram</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"hg19"</span>,plot.space <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></code></pre></div>
<p><img src="../man/figures/README-ideogram_coverage_1-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">basic.coverage</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_transcript.html">geom_transcript</a></span><span class="op">(</span>gtf.gr<span class="op">=</span><span class="va">gtf.gr</span>,label.vjust <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_ideogram.html">geom_ideogram</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"hg19"</span>,plot.space <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></code></pre></div>
<p><img src="../man/figures/README-ideogram_coverage_2-1.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="dna-seq-data" class="section level1">
<h1 class="hasAnchor">
<a href="#dna-seq-data" class="anchor"></a>DNA-seq data</h1>
<div id="cnv" class="section level2">
<h2 class="hasAnchor">
<a href="#cnv" class="anchor"></a>CNV</h2>
<div id="load-the-data-1" class="section level3">
<h3 class="hasAnchor">
<a href="#load-the-data-1" class="anchor"></a>Load the data</h3>
<p>The DNA-seq data used here are from <a href="http://bioconductor.org/help/course-materials/2014/SeattleOct2014/B02.2.3_CopyNumber.html">Copy number work flow</a>, we select tumor sample, and get bin counts with <code><a href="https://rdrr.io/pkg/cn.mops/man/getReadCountsFromBAM.html">cn.mops::getReadCountsFromBAM</a></code> with <code>WL</code> 1000.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># track file</span>
<span class="va">track.file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"DNA-seq"</span>, <span class="st">"CNV_example.txt"</span>, package <span class="op">=</span> <span class="st">"ggcoverage"</span><span class="op">)</span>
<span class="va">track.df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="va">track.file</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co"># check data</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">track.df</span><span class="op">)</span>
<span class="co">#&gt;   seqnames    start      end score  Type Group</span>
<span class="co">#&gt; 1     chr4 61743001 61744000    17 tumor tumor</span>
<span class="co">#&gt; 2     chr4 61744001 61745000    14 tumor tumor</span>
<span class="co">#&gt; 3     chr4 61745001 61746000    13 tumor tumor</span>
<span class="co">#&gt; 4     chr4 61746001 61747000    16 tumor tumor</span>
<span class="co">#&gt; 5     chr4 61747001 61748000    25 tumor tumor</span>
<span class="co">#&gt; 6     chr4 61748001 61749000    24 tumor tumor</span></code></pre></div>
</div>
<div id="basic-coverage-1" class="section level3">
<h3 class="hasAnchor">
<a href="#basic-coverage-1" class="anchor"></a>Basic coverage</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">basic.coverage</span> <span class="op">=</span> <span class="fu"><a href="../reference/ggcoverage.html">ggcoverage</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">track.df</span>,color <span class="op">=</span> <span class="cn">NULL</span>, mark.region <span class="op">=</span> <span class="cn">NULL</span>,
                            region <span class="op">=</span> <span class="st">'chr4:61750000-62,700,000'</span>, range.position <span class="op">=</span> <span class="st">"out"</span><span class="op">)</span>
<span class="va">basic.coverage</span></code></pre></div>
<p><img src="../man/figures/README-basic_coverage_dna-1.png" width="576" style="display: block; margin: auto;"></p>
</div>
<div id="add-annotations" class="section level3">
<h3 class="hasAnchor">
<a href="#add-annotations" class="anchor"></a>Add annotations</h3>
<p>Add <strong>GC</strong>, <strong>ideogram</strong> and <strong>gene</strong> annotations.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load genome data</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"BSgenome.Hsapiens.UCSC.hg19"</span><span class="op">)</span>
<span class="co"># create plot</span>
<span class="va">basic.coverage</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_gc.html">geom_gc</a></span><span class="op">(</span>bs.fa.seq<span class="op">=</span><span class="va">BSgenome.Hsapiens.UCSC.hg19</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_gene.html">geom_gene</a></span><span class="op">(</span>gtf.gr<span class="op">=</span><span class="va">gtf.gr</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_ideogram.html">geom_ideogram</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"hg19"</span><span class="op">)</span></code></pre></div>
<p><img src="../man/figures/README-gc_coverage-1.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="single-nucleotide-level" class="section level2">
<h2 class="hasAnchor">
<a href="#single-nucleotide-level" class="anchor"></a>Single-nucleotide level</h2>
<div id="load-the-data-2" class="section level3">
<h3 class="hasAnchor">
<a href="#load-the-data-2" class="anchor"></a>Load the data</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># prepare sample metadata</span>
<span class="va">sample.meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
  SampleName <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"tumorA.chr4.selected"</span><span class="op">)</span>,
  Type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"tumorA"</span><span class="op">)</span>,
  Group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"tumorA"</span><span class="op">)</span>
<span class="op">)</span>
<span class="co"># load bam file</span>
<span class="va">bam.file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"DNA-seq"</span>, <span class="st">"tumorA.chr4.selected.bam"</span>, package <span class="op">=</span> <span class="st">"ggcoverage"</span><span class="op">)</span>
<span class="va">track.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/LoadTrackFile.html">LoadTrackFile</a></span><span class="op">(</span>
  track.file <span class="op">=</span> <span class="va">bam.file</span>,
  meta.info <span class="op">=</span> <span class="va">sample.meta</span>,
  single.nuc<span class="op">=</span><span class="cn">TRUE</span>, single.nuc.region<span class="op">=</span><span class="st">"chr4:62474235-62474295"</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">track.df</span><span class="op">)</span>
<span class="co">#&gt;   seqnames    start      end score   Type  Group</span>
<span class="co">#&gt; 1     chr4 62474235 62474236     5 tumorA tumorA</span>
<span class="co">#&gt; 2     chr4 62474236 62474237     5 tumorA tumorA</span>
<span class="co">#&gt; 3     chr4 62474237 62474238     5 tumorA tumorA</span>
<span class="co">#&gt; 4     chr4 62474238 62474239     6 tumorA tumorA</span>
<span class="co">#&gt; 5     chr4 62474239 62474240     6 tumorA tumorA</span>
<span class="co">#&gt; 6     chr4 62474240 62474241     6 tumorA tumorA</span></code></pre></div>
</div>
<div id="default-color-scheme" class="section level3">
<h3 class="hasAnchor">
<a href="#default-color-scheme" class="anchor"></a>Default color scheme</h3>
<p>For base and amino acid annotation, we have following default color schemes, you can change with <code>nuc.color</code> and <code>aa.color</code> parameters.</p>
<p>Default color scheme for base annotation is <code>Clustal-style</code>, more popular color schemes is available <a href="https://www.biostars.org/p/171056/">here</a>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># color scheme</span>
<span class="va">nuc.color</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"A"</span> <span class="op">=</span> <span class="st">"#ff2b08"</span>, <span class="st">"C"</span> <span class="op">=</span> <span class="st">"#009aff"</span>, <span class="st">"G"</span> <span class="op">=</span> <span class="st">"#ffb507"</span>, <span class="st">"T"</span> <span class="op">=</span> <span class="st">"#00bc0d"</span><span class="op">)</span>
<span class="va">opar</span> <span class="op">&lt;-</span> <span class="fu">graphics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span><span class="op">)</span> 
<span class="co"># create plot</span>
<span class="fu">graphics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">5</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span>
<span class="fu">graphics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/graphics/image.html">image</a></span><span class="op">(</span>
  <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">nuc.color</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">nuc.color</span><span class="op">)</span><span class="op">)</span>,
  col <span class="op">=</span> <span class="va">nuc.color</span>,
  xlab <span class="op">=</span> <span class="st">""</span>, ylab <span class="op">=</span> <span class="st">""</span>, xaxt <span class="op">=</span> <span class="st">"n"</span>, yaxt <span class="op">=</span> <span class="st">"n"</span>, bty <span class="op">=</span> <span class="st">"n"</span>
<span class="op">)</span>
<span class="fu">graphics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">nuc.color</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">nuc.color</span><span class="op">)</span><span class="op">)</span>
<span class="fu">graphics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>
  text <span class="op">=</span> <span class="st">"Base"</span>, adj <span class="op">=</span> <span class="fl">1</span>, las <span class="op">=</span> <span class="fl">1</span>,
  side <span class="op">=</span> <span class="fl">2</span>
<span class="op">)</span></code></pre></div>
<p><img src="ggcoverage_files/figure-html/base_color_scheme-1.png" width="360" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># reset par default</span>
<span class="fu">graphics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span><span class="va">opar</span><span class="op">)</span></code></pre></div>
<p>Default color scheme for amino acid annotation is from <a href="https://academic.oup.com/peds/article/10/7/743/1593029?login=false">Residual colours: a proposal for aminochromography</a>:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">aa.color</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
  <span class="st">"D"</span> <span class="op">=</span> <span class="st">"#FF0000"</span>, <span class="st">"S"</span> <span class="op">=</span> <span class="st">"#FF2400"</span>, <span class="st">"T"</span> <span class="op">=</span> <span class="st">"#E34234"</span>, <span class="st">"G"</span> <span class="op">=</span> <span class="st">"#FF8000"</span>, <span class="st">"P"</span> <span class="op">=</span> <span class="st">"#F28500"</span>,
  <span class="st">"C"</span> <span class="op">=</span> <span class="st">"#FFFF00"</span>, <span class="st">"A"</span> <span class="op">=</span> <span class="st">"#FDFF00"</span>, <span class="st">"V"</span> <span class="op">=</span> <span class="st">"#E3FF00"</span>, <span class="st">"I"</span> <span class="op">=</span> <span class="st">"#C0FF00"</span>, <span class="st">"L"</span> <span class="op">=</span> <span class="st">"#89318C"</span>,
  <span class="st">"M"</span> <span class="op">=</span> <span class="st">"#00FF00"</span>, <span class="st">"F"</span> <span class="op">=</span> <span class="st">"#50C878"</span>, <span class="st">"Y"</span> <span class="op">=</span> <span class="st">"#30D5C8"</span>, <span class="st">"W"</span> <span class="op">=</span> <span class="st">"#00FFFF"</span>, <span class="st">"H"</span> <span class="op">=</span> <span class="st">"#0F2CB3"</span>,
  <span class="st">"R"</span> <span class="op">=</span> <span class="st">"#0000FF"</span>, <span class="st">"K"</span> <span class="op">=</span> <span class="st">"#4b0082"</span>, <span class="st">"N"</span> <span class="op">=</span> <span class="st">"#800080"</span>, <span class="st">"Q"</span> <span class="op">=</span> <span class="st">"#FF00FF"</span>, <span class="st">"E"</span> <span class="op">=</span> <span class="st">"#8F00FF"</span>,
  <span class="st">"*"</span> <span class="op">=</span> <span class="st">"#FFC0CB"</span>, <span class="st">" "</span> <span class="op">=</span> <span class="st">"#FFFFFF"</span>, <span class="st">" "</span> <span class="op">=</span> <span class="st">"#FFFFFF"</span>, <span class="st">" "</span> <span class="op">=</span> <span class="st">"#FFFFFF"</span>, <span class="st">" "</span> <span class="op">=</span> <span class="st">"#FFFFFF"</span>
<span class="op">)</span>

<span class="fu">graphics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">5</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span>
<span class="fu">graphics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/graphics/image.html">image</a></span><span class="op">(</span>
  <span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">aa.color</span><span class="op">)</span>,nrow<span class="op">=</span><span class="fl">5</span><span class="op">)</span>,
  col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html">rev</a></span><span class="op">(</span><span class="va">aa.color</span><span class="op">)</span>,
  xlab <span class="op">=</span> <span class="st">""</span>, ylab <span class="op">=</span> <span class="st">""</span>, xaxt <span class="op">=</span> <span class="st">"n"</span>, yaxt <span class="op">=</span> <span class="st">"n"</span>, bty <span class="op">=</span> <span class="st">"n"</span>
<span class="op">)</span>
<span class="fu">graphics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/expand.grid.html">expand.grid</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rev.html">rev</a></span><span class="op">(</span><span class="va">aa.color</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu">graphics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>
  text <span class="op">=</span> <span class="st">"Amino acids"</span>, adj <span class="op">=</span> <span class="fl">1</span>, las <span class="op">=</span> <span class="fl">1</span>,
  side <span class="op">=</span> <span class="fl">2</span>
<span class="op">)</span></code></pre></div>
<p><img src="ggcoverage_files/figure-html/aa_color_scheme-1.png" width="600" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># reset par default</span>
<span class="fu">graphics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span><span class="va">opar</span><span class="op">)</span></code></pre></div>
</div>
<div id="add-base-and-amino-acid-annotation" class="section level3">
<h3 class="hasAnchor">
<a href="#add-base-and-amino-acid-annotation" class="anchor"></a>Add base and amino acid annotation</h3>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/ggcoverage.html">ggcoverage</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">track.df</span>, color <span class="op">=</span> <span class="st">"grey"</span>, range.position <span class="op">=</span> <span class="st">"out"</span>, single.nuc<span class="op">=</span><span class="cn">T</span>, rect.color <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_base.html">geom_base</a></span><span class="op">(</span>bam.file <span class="op">=</span> <span class="va">bam.file</span>,
            bs.fa.seq <span class="op">=</span> <span class="va">BSgenome.Hsapiens.UCSC.hg19</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_ideogram.html">geom_ideogram</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"hg19"</span>,plot.space <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></code></pre></div>
<p><img src="../man/figures/README-base_aa_coverage-1.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
</div>
<div id="chip-seq-data" class="section level1">
<h1 class="hasAnchor">
<a href="#chip-seq-data" class="anchor"></a>ChIP-seq data</h1>
<p>The ChIP-seq data used here are from <a href="https://bioconductor.org/packages/release/bioc/html/DiffBind.html">DiffBind</a>, I select four sample to use as example: Chr18_MCF7_input, Chr18_MCF7_ER_1, Chr18_MCF7_ER_3, Chr18_MCF7_ER_2, and all bam files are converted to bigwig file with <a href="https://deeptools.readthedocs.io/en/develop/">deeptools</a>.</p>
<p>Create metadata:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load metadata</span>
<span class="va">sample.meta</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>SampleName<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Chr18_MCF7_ER_1'</span>,<span class="st">'Chr18_MCF7_ER_2'</span>,<span class="st">'Chr18_MCF7_ER_3'</span>,<span class="st">'Chr18_MCF7_input'</span><span class="op">)</span>,
                         Type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"MCF7_ER_1"</span>,<span class="st">"MCF7_ER_2"</span>,<span class="st">"MCF7_ER_3"</span>,<span class="st">"MCF7_input"</span><span class="op">)</span>,
                         Group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"IP"</span>, <span class="st">"IP"</span>, <span class="st">"IP"</span>, <span class="st">"Input"</span><span class="op">)</span><span class="op">)</span>
<span class="va">sample.meta</span>
<span class="co">#&gt;         SampleName       Type Group</span>
<span class="co">#&gt; 1  Chr18_MCF7_ER_1  MCF7_ER_1    IP</span>
<span class="co">#&gt; 2  Chr18_MCF7_ER_2  MCF7_ER_2    IP</span>
<span class="co">#&gt; 3  Chr18_MCF7_ER_3  MCF7_ER_3    IP</span>
<span class="co">#&gt; 4 Chr18_MCF7_input MCF7_input Input</span></code></pre></div>
<p>Load track files:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># track folder</span>
<span class="va">track.folder</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"ChIP-seq"</span>, package <span class="op">=</span> <span class="st">"ggcoverage"</span><span class="op">)</span>
<span class="co"># load bigwig file</span>
<span class="va">track.df</span> <span class="op">=</span> <span class="fu"><a href="../reference/LoadTrackFile.html">LoadTrackFile</a></span><span class="op">(</span>track.folder <span class="op">=</span> <span class="va">track.folder</span>, format <span class="op">=</span> <span class="st">"bw"</span>,
                         meta.info <span class="op">=</span> <span class="va">sample.meta</span><span class="op">)</span>
<span class="co"># check data</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">track.df</span><span class="op">)</span>
<span class="co">#&gt;   seqnames    start      end   score      Type Group</span>
<span class="co">#&gt; 1    chr18 76799701 76800000 439.316 MCF7_ER_1    IP</span>
<span class="co">#&gt; 2    chr18 76800001 76800300 658.974 MCF7_ER_1    IP</span>
<span class="co">#&gt; 3    chr18 76800301 76800600 219.658 MCF7_ER_1    IP</span>
<span class="co">#&gt; 4    chr18 76800601 76800900 658.974 MCF7_ER_1    IP</span>
<span class="co">#&gt; 5    chr18 76800901 76801200   0.000 MCF7_ER_1    IP</span>
<span class="co">#&gt; 6    chr18 76801201 76801500 219.658 MCF7_ER_1    IP</span></code></pre></div>
<p>Prepare mark region:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># create mark region</span>
<span class="va">mark.region</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>start<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">76822533</span><span class="op">)</span>,
                       end<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">76823743</span><span class="op">)</span>,
                       label<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Promoter"</span><span class="op">)</span><span class="op">)</span>
<span class="co"># check data</span>
<span class="va">mark.region</span>
<span class="co">#&gt;      start      end    label</span>
<span class="co">#&gt; 1 76822533 76823743 Promoter</span></code></pre></div>
<hr>
<div id="basic-coverage-2" class="section level2">
<h2 class="hasAnchor">
<a href="#basic-coverage-2" class="anchor"></a>Basic coverage</h2>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">basic.coverage</span> <span class="op">=</span> <span class="fu"><a href="../reference/ggcoverage.html">ggcoverage</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">track.df</span>, color <span class="op">=</span> <span class="st">"auto"</span>, region <span class="op">=</span> <span class="st">"chr18:76822285-76900000"</span>, 
                            mark.region<span class="op">=</span><span class="va">mark.region</span>, show.mark.label <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">basic.coverage</span></code></pre></div>
<p><img src="../man/figures/README-basic_coverage_chip-1.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
<div id="add-annotations-1" class="section level2">
<h2 class="hasAnchor">
<a href="#add-annotations-1" class="anchor"></a>Add annotations</h2>
<p>Add <strong>gene</strong>, <strong>ideogram</strong> and <strong>peak</strong> annotations. To create peak annotation, we first <strong>get consensus peaks</strong> with <a href="https://github.com/Genometric/MSPC">MSPC</a>, you can also use DEbChIP’s <code>GetConsensusPeak</code> (<code>MSPC</code>’s wrapper) to do this.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get consensus peak file</span>
<span class="va">peak.file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"ChIP-seq"</span>, <span class="st">"consensus.peak"</span>, package <span class="op">=</span> <span class="st">"ggcoverage"</span><span class="op">)</span>

<span class="va">basic.coverage</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_gene.html">geom_gene</a></span><span class="op">(</span>gtf.gr<span class="op">=</span><span class="va">gtf.gr</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_peak.html">geom_peak</a></span><span class="op">(</span>bed.file <span class="op">=</span> <span class="va">peak.file</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_ideogram.html">geom_ideogram</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"hg19"</span>,plot.space <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></code></pre></div>
<p><img src="../man/figures/README-peak_coverage-1.png" width="576" style="display: block; margin: auto;"></p>
<hr>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.3 (2020-10-10)</span>
<span class="co">#&gt; Platform: x86_64-conda-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: CentOS Linux 7 (Core)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS/LAPACK: /home/softwares/anaconda3/envs/r4.0/lib/libopenblasp-r0.3.12.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=zh_CN.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=zh_CN.UTF-8        LC_COLLATE=zh_CN.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=zh_CN.UTF-8    LC_MESSAGES=zh_CN.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=zh_CN.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=zh_CN.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] ggcoverage_0.7.0     rtracklayer_1.50.0   GenomicRanges_1.42.0</span>
<span class="co">#&gt; [4] GenomeInfoDb_1.26.7  IRanges_2.24.1       S4Vectors_0.28.1    </span>
<span class="co">#&gt; [7] BiocGenerics_0.42.0  knitr_1.37           BiocStyle_2.18.1    </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] backports_1.2.1             Hmisc_4.6-0                </span>
<span class="co">#&gt;   [3] BiocFileCache_1.14.0        systemfonts_1.0.1          </span>
<span class="co">#&gt;   [5] plyr_1.8.6                  lazyeval_0.2.2             </span>
<span class="co">#&gt;   [7] splines_4.0.3               BiocParallel_1.24.1        </span>
<span class="co">#&gt;   [9] ggplot2_3.3.5               digest_0.6.27              </span>
<span class="co">#&gt;  [11] ensembldb_2.14.1            htmltools_0.5.2            </span>
<span class="co">#&gt;  [13] fansi_0.4.2                 magrittr_2.0.1             </span>
<span class="co">#&gt;  [15] checkmate_2.0.0             memoise_2.0.0              </span>
<span class="co">#&gt;  [17] BSgenome_1.58.0             cluster_2.1.1              </span>
<span class="co">#&gt;  [19] Biostrings_2.58.0           matrixStats_0.58.0         </span>
<span class="co">#&gt;  [21] ggbio_1.38.0                askpass_1.1                </span>
<span class="co">#&gt;  [23] pkgdown_1.6.1               prettyunits_1.1.1          </span>
<span class="co">#&gt;  [25] ggh4x_0.2.1                 jpeg_0.1-8.1               </span>
<span class="co">#&gt;  [27] colorspace_2.0-0            blob_1.2.1                 </span>
<span class="co">#&gt;  [29] rappdirs_0.3.3              ggrepel_0.9.1              </span>
<span class="co">#&gt;  [31] textshaping_0.1.2           xfun_0.30                  </span>
<span class="co">#&gt;  [33] dplyr_1.0.5                 crayon_1.4.1               </span>
<span class="co">#&gt;  [35] RCurl_1.98-1.3              jsonlite_1.7.2             </span>
<span class="co">#&gt;  [37] graph_1.68.0                survival_3.2-10            </span>
<span class="co">#&gt;  [39] VariantAnnotation_1.36.0    glue_1.6.2                 </span>
<span class="co">#&gt;  [41] gtable_0.3.0                zlibbioc_1.36.0            </span>
<span class="co">#&gt;  [43] XVector_0.30.0              seqinr_4.2-5               </span>
<span class="co">#&gt;  [45] DelayedArray_0.16.3         scales_1.1.1               </span>
<span class="co">#&gt;  [47] DBI_1.1.1                   GGally_2.1.2               </span>
<span class="co">#&gt;  [49] Rcpp_1.0.6                  progress_1.2.2             </span>
<span class="co">#&gt;  [51] htmlTable_2.4.0             foreign_0.8-81             </span>
<span class="co">#&gt;  [53] bit_4.0.4                   OrganismDbi_1.32.0         </span>
<span class="co">#&gt;  [55] Formula_1.2-4               htmlwidgets_1.5.3          </span>
<span class="co">#&gt;  [57] httr_1.4.2                  RColorBrewer_1.1-2         </span>
<span class="co">#&gt;  [59] ellipsis_0.3.2              pkgconfig_2.0.3            </span>
<span class="co">#&gt;  [61] reshape_0.8.8               XML_3.99-0.6               </span>
<span class="co">#&gt;  [63] nnet_7.3-16                 sass_0.4.1                 </span>
<span class="co">#&gt;  [65] dbplyr_2.1.1                utf8_1.2.1                 </span>
<span class="co">#&gt;  [67] tidyselect_1.1.0            rlang_1.0.3                </span>
<span class="co">#&gt;  [69] reshape2_1.4.4              AnnotationDbi_1.52.0       </span>
<span class="co">#&gt;  [71] munsell_0.5.0               tools_4.0.3                </span>
<span class="co">#&gt;  [73] cachem_1.0.4                cli_3.3.0                  </span>
<span class="co">#&gt;  [75] generics_0.1.0              RSQLite_2.2.5              </span>
<span class="co">#&gt;  [77] ade4_1.7-16                 evaluate_0.14              </span>
<span class="co">#&gt;  [79] stringr_1.4.0               fastmap_1.1.0              </span>
<span class="co">#&gt;  [81] yaml_2.2.1                  ragg_0.4.0                 </span>
<span class="co">#&gt;  [83] bit64_4.0.5                 fs_1.5.0                   </span>
<span class="co">#&gt;  [85] purrr_0.3.4                 AnnotationFilter_1.14.0    </span>
<span class="co">#&gt;  [87] RBGL_1.66.0                 xml2_1.3.2                 </span>
<span class="co">#&gt;  [89] biomaRt_2.46.3              compiler_4.0.3             </span>
<span class="co">#&gt;  [91] rstudioapi_0.13             curl_4.3                   </span>
<span class="co">#&gt;  [93] png_0.1-7                   tibble_3.1.0               </span>
<span class="co">#&gt;  [95] bslib_0.3.1                 stringi_1.5.3              </span>
<span class="co">#&gt;  [97] highr_0.8                   GenomicFeatures_1.42.2     </span>
<span class="co">#&gt;  [99] desc_1.3.0                  lattice_0.20-45            </span>
<span class="co">#&gt; [101] ProtGenerics_1.22.0         Matrix_1.3-3               </span>
<span class="co">#&gt; [103] vctrs_0.4.1                 pillar_1.5.1               </span>
<span class="co">#&gt; [105] lifecycle_1.0.0             BiocManager_1.30.16        </span>
<span class="co">#&gt; [107] jquerylib_0.1.3             data.table_1.14.2          </span>
<span class="co">#&gt; [109] bitops_1.0-6                patchwork_1.0.0            </span>
<span class="co">#&gt; [111] R6_2.5.0                    latticeExtra_0.6-29        </span>
<span class="co">#&gt; [113] bookdown_0.26               gridExtra_2.3              </span>
<span class="co">#&gt; [115] dichromat_2.0-0             MASS_7.3-53.1              </span>
<span class="co">#&gt; [117] assertthat_0.2.1            SummarizedExperiment_1.20.0</span>
<span class="co">#&gt; [119] openssl_1.4.3               rprojroot_2.0.2            </span>
<span class="co">#&gt; [121] GenomicAlignments_1.26.0    Rsamtools_2.6.0            </span>
<span class="co">#&gt; [123] GenomeInfoDbData_1.2.4      parallel_4.0.3             </span>
<span class="co">#&gt; [125] hms_1.0.0                   grid_4.0.3                 </span>
<span class="co">#&gt; [127] rpart_4.1-15                rmarkdown_2.14             </span>
<span class="co">#&gt; [129] MatrixGenerics_1.2.1        biovizBase_1.38.0          </span>
<span class="co">#&gt; [131] Biobase_2.50.0              base64enc_0.1-3</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
