<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Add Base and Amino Acid Annotation to Coverage Plot. — geom_base • ggcoverage</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Add Base and Amino Acid Annotation to Coverage Plot. — geom_base" />
<meta property="og:description" content="Add Base and Amino Acid Annotation to Coverage Plot." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ggcoverage</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.7.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/ggcoverage.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Add Base and Amino Acid Annotation to Coverage Plot.</h1>
    
    <div class="hidden name"><code>geom_base.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Add Base and Amino Acid Annotation to Coverage Plot.</p>
    </div>

    <pre class="usage"><span class='fu'>geom_base</span><span class='op'>(</span>
  <span class='va'>bam.file</span>,
  fa.file <span class='op'>=</span> <span class='cn'>NULL</span>,
  bs.fa.seq <span class='op'>=</span> <span class='cn'>NULL</span>,
  chr.split <span class='op'>=</span> <span class='st'>"[[:space:]]"</span>,
  nuc.offset <span class='op'>=</span> <span class='op'>-</span><span class='fl'>0.1</span>,
  nuc.size <span class='op'>=</span> <span class='fl'>4</span>,
  nuc.padding <span class='op'>=</span> <span class='fl'>0.05</span>,
  nuc.padding.r <span class='op'>=</span> <span class='fl'>0</span>,
  nuc.color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>A <span class='op'>=</span> <span class='st'>"#ff2b08"</span>, C <span class='op'>=</span> <span class='st'>"#009aff"</span>, G <span class='op'>=</span> <span class='st'>"#ffb507"</span>, T <span class='op'>=</span> <span class='st'>"#00bc0d"</span><span class='op'>)</span>,
  guide.line <span class='op'>=</span> <span class='cn'>NULL</span>,
  guide.line.color <span class='op'>=</span> <span class='st'>"red"</span>,
  guide.line.type <span class='op'>=</span> <span class='st'>"dashed"</span>,
  show.aa <span class='op'>=</span> <span class='cn'>TRUE</span>,
  sens <span class='op'>=</span> <span class='st'>"F"</span>,
  numcode <span class='op'>=</span> <span class='fl'>1</span>,
  NAstring <span class='op'>=</span> <span class='st'>"X"</span>,
  ambiguous <span class='op'>=</span> <span class='cn'>FALSE</span>,
  aa.color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>D <span class='op'>=</span> <span class='st'>"#FF0000"</span>, S <span class='op'>=</span> <span class='st'>"#FF2400"</span>, T <span class='op'>=</span> <span class='st'>"#E34234"</span>, G <span class='op'>=</span> <span class='st'>"#FF8000"</span>, P <span class='op'>=</span>
    <span class='st'>"#F28500"</span>, C <span class='op'>=</span> <span class='st'>"#FFFF00"</span>, A <span class='op'>=</span> <span class='st'>"#FDFF00"</span>, V <span class='op'>=</span> <span class='st'>"#E3FF00"</span>, I <span class='op'>=</span> <span class='st'>"#C0FF00"</span>, L <span class='op'>=</span> <span class='st'>"#89318C"</span>,
    M <span class='op'>=</span> <span class='st'>"#00FF00"</span>, F <span class='op'>=</span> <span class='st'>"#50C878"</span>, Y <span class='op'>=</span> <span class='st'>"#30D5C8"</span>, W <span class='op'>=</span> <span class='st'>"#00FFFF"</span>, H <span class='op'>=</span> <span class='st'>"#0F2CB3"</span>, R <span class='op'>=</span>
    <span class='st'>"#0000FF"</span>, K <span class='op'>=</span> <span class='st'>"#4b0082"</span>, N <span class='op'>=</span> <span class='st'>"#800080"</span>, Q <span class='op'>=</span> <span class='st'>"#FF00FF"</span>, E <span class='op'>=</span> <span class='st'>"#8F00FF"</span>, `*` <span class='op'>=</span>
    <span class='st'>"#FFC0CB"</span><span class='op'>)</span>,
  aa.size <span class='op'>=</span> <span class='fl'>4</span>,
  aa.margin <span class='op'>=</span> <span class='fl'>2</span>,
  aa.height <span class='op'>=</span> <span class='fl'>0.4</span>,
  plot.space <span class='op'>=</span> <span class='fl'>2.5</span>,
  plot.height <span class='op'>=</span> <span class='fl'>0.5</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>bam.file</th>
      <td><p>BAM file.</p></td>
    </tr>
    <tr>
      <th>fa.file</th>
      <td><p>Genome fasta file. Default: NULL.</p></td>
    </tr>
    <tr>
      <th>bs.fa.seq</th>
      <td><p>BSgenome for species. Default: NULL.</p></td>
    </tr>
    <tr>
      <th>chr.split</th>
      <td><p>Split between chromosome name and description in <code>fa.file</code>. Default: "[[:space:]]".</p></td>
    </tr>
    <tr>
      <th>nuc.offset</th>
      <td><p>Offset of nucleotide to frequency plot. Default: -0.1.</p></td>
    </tr>
    <tr>
      <th>nuc.size</th>
      <td><p>The size of nucleotide text. Default: 4.</p></td>
    </tr>
    <tr>
      <th>nuc.padding</th>
      <td><p>Background padding of nucleotide annotation. Default: 0.05.</p></td>
    </tr>
    <tr>
      <th>nuc.padding.r</th>
      <td><p>Radius of background padding. Default: 0.</p></td>
    </tr>
    <tr>
      <th>nuc.color</th>
      <td><p>Color scheme for nucleotides. Default: "A": "#ff2b08", "C": "#009aff",
"G": "#ffb507", "T": "#00bc0d".</p></td>
    </tr>
    <tr>
      <th>guide.line</th>
      <td><p>Nucleotide frequency guide line. Default: NULL (0.5).</p></td>
    </tr>
    <tr>
      <th>guide.line.color</th>
      <td><p>The color of guide line. Default: "red".</p></td>
    </tr>
    <tr>
      <th>guide.line.type</th>
      <td><p>The line type of guide line. Default: "dashed".</p></td>
    </tr>
    <tr>
      <th>show.aa</th>
      <td><p>Logical value, whether to show amino acid. Default: TRUE.</p></td>
    </tr>
    <tr>
      <th>sens</th>
      <td><p>Sense to translate: F for forward sense and R for reverse sense.
Parameter of <code>translate</code>. Default: F.</p></td>
    </tr>
    <tr>
      <th>numcode</th>
      <td><p>The ncbi genetic code number for translation.
Parameter of <code>translate</code>. By default the standard genetic code is used.</p></td>
    </tr>
    <tr>
      <th>NAstring</th>
      <td><p>How to translate amino-acids when there are ambiguous bases in codons.
Parameter of <code>translate</code>. Default: X.</p></td>
    </tr>
    <tr>
      <th>ambiguous</th>
      <td><p>If TRUE, ambiguous bases are taken into account so that for instance GGN is
translated to Gly in the standard genetic code. Parameter of <code>translate</code>. Default: FALSE.</p></td>
    </tr>
    <tr>
      <th>aa.color</th>
      <td><p>Color scheme for amino acids.</p></td>
    </tr>
    <tr>
      <th>aa.size</th>
      <td><p>The size of amino acid text. Default: 4.</p></td>
    </tr>
    <tr>
      <th>aa.margin</th>
      <td><p>Top and bottom margin of amino acids. Default: 2.</p></td>
    </tr>
    <tr>
      <th>aa.height</th>
      <td><p>The relative height of amino acid to base frequency plot. Default: 0.4.</p></td>
    </tr>
    <tr>
      <th>plot.space</th>
      <td><p>Top and bottom margin. Default: 2.5.</p></td>
    </tr>
    <tr>
      <th>plot.height</th>
      <td><p>The relative height of base and amino acid annotation to coverage plot. Default: 0.5.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Plot.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>ggcoverage</span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='st'>"BSgenome.Hsapiens.UCSC.hg19"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>载入需要的程辑包：BSgenome</span></div><div class='output co'>#&gt; <span class='message'>载入需要的程辑包：BiocGenerics</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>载入程辑包：'BiocGenerics'</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:stats':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    IQR, mad, sd, var, xtabs</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:base':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
#&gt; <span class='message'>    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
#&gt; <span class='message'>    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
#&gt; <span class='message'>    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
#&gt; <span class='message'>    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
#&gt; <span class='message'>    union, unique, unsplit, which.max, which.min</span></div><div class='output co'>#&gt; <span class='message'>载入需要的程辑包：S4Vectors</span></div><div class='output co'>#&gt; <span class='message'>载入需要的程辑包：stats4</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>载入程辑包：'S4Vectors'</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from 'package:base':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand.grid</span></div><div class='output co'>#&gt; <span class='message'>载入需要的程辑包：IRanges</span></div><div class='output co'>#&gt; <span class='message'>载入需要的程辑包：GenomeInfoDb</span></div><div class='output co'>#&gt; <span class='message'>载入需要的程辑包：GenomicRanges</span></div><div class='output co'>#&gt; <span class='message'>载入需要的程辑包：Biostrings</span></div><div class='output co'>#&gt; <span class='message'>载入需要的程辑包：XVector</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>载入程辑包：'Biostrings'</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from 'package:base':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    strsplit</span></div><div class='output co'>#&gt; <span class='message'>载入需要的程辑包：rtracklayer</span></div><div class='input'><span class='co'># get sample metadata</span>
<span class='va'>sample.meta</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>
  SampleName <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"tumorA.chr4.selected"</span><span class='op'>)</span>,
  Type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"tumorA"</span><span class='op'>)</span>, Group <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"tumorA"</span><span class='op'>)</span>
<span class='op'>)</span>
<span class='co'># get bam file</span>
<span class='va'>bam.file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"DNA-seq"</span>, <span class='st'>"tumorA.chr4.selected.bam"</span>, package <span class='op'>=</span> <span class='st'>"ggcoverage"</span><span class='op'>)</span>
<span class='co'># load bam file</span>
<span class='va'>track.df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='LoadTrackFile.html'>LoadTrackFile</a></span><span class='op'>(</span>
  track.file <span class='op'>=</span> <span class='va'>bam.file</span>,
  meta.info <span class='op'>=</span> <span class='va'>sample.meta</span>, single.nuc <span class='op'>=</span> <span class='cn'>TRUE</span>,
  single.nuc.region <span class='op'>=</span> <span class='st'>"chr4:62474235-62474295"</span>
<span class='op'>)</span>
<span class='fu'><a href='ggcoverage.html'>ggcoverage</a></span><span class='op'>(</span>
  data <span class='op'>=</span> <span class='va'>track.df</span>, color <span class='op'>=</span> <span class='st'>"grey"</span>, range.position <span class='op'>=</span> <span class='st'>"out"</span>,
  single.nuc <span class='op'>=</span> <span class='cn'>TRUE</span>, rect.color <span class='op'>=</span> <span class='st'>"white"</span>
<span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_base</span><span class='op'>(</span>
    bam.file <span class='op'>=</span> <span class='va'>bam.file</span>,
    bs.fa.seq <span class='op'>=</span> <span class='va'>BSgenome.Hsapiens.UCSC.hg19</span>
  <span class='op'>)</span>
</div><div class='img'><img src='geom_base-1.png' alt='' width='700' height='433' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Yabing Song.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


